device/iot-device-samples/send-event/src/main/java/samples/com/microsoft/azure/sdk/iot/InputParameters.java

```diff
@@ -0,0 +1,139 @@
+
+    private static final String FOOTER = "\nFor more info, please refer to https://github.com/Azure/azure-iot-sdks";
+    private static final String APPEXE = "java -jar ";
+
+    private static CommandLine cmd = null;
+
+    /**
+     * Setup parameters from command line arguments
+     * @param args array from main()
+     */
+    public InputParameters(String[] args){
+
+        // create option for cli input
+        Options options = new Options()
+                .addOption(
+                        Option.builder("h")
+                                .longOpt("help")
+                                .hasArg(false)
+                                .desc("Prints help message")
+                                .build()
+                )
+                .addOption(
+                        Option.builder("c")
+                                .longOpt("connectionString")
+                                .hasArg()
+                                .desc("IoT Hub or Edge Hub connection string (required; contains Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key> or HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>;GatewayHostName=<gateway>)")
+                                .build()
+                )
+                .addOption(
+                        Option.builder("r")
+                                .longOpt("requests")
+                                .hasArg()
+                                .desc("Number of requests to send (required)")
+                                .build()
+                )
+                .addOption(
+                        Option.builder("p")
+                                .longOpt("protocol")
+                                .hasArg()
+                                .desc("Protocol [mqtt | https | amqps | amqps_ws | mqtt_ws] (optional; defaults to \"mqtt\")")
+                                .build()
+                )
+                .addOption(
+                        Option.builder("pc")
+                                .longOpt("pathToCertificate")
+                                .hasArg()
+                                .desc("Path to certificate to enable one-way authentication over ssl (not necessary when connecting directly to Iot Hub, but required if connecting to an Edge device using a non public root CA certificate)")
+                                .build()
+                );
+
+        String cmdLine = APPEXE;
+        try {
+            String jarPath = Options.class
+                    .getProtectionDomain()
+                    .getCodeSource()
+                    .getLocation()
+                    .toURI()
+                    .getPath();
+            String jarName = jarPath.substring(jarPath.lastIndexOf("/") + 1);
+            cmdLine += jarName;
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
```

Comment:
This will print the stacktrace to the output stream. We might not want to do that as it might be confusing if when getting help the program throws an exception.

Reply:
> this exception is to let users aware of illegal syntax while parsing the URL, for the jar file path specifically. To avoid of possible confusion here, I can print out a simple sentence instead of full stack trace. Thoughts?

Reply:
This is a better as it doesn't print a stack trace. However, the customer isn't supplying any parameter here. We are using reflection to get the the URI of the JAR and using it in the example string. If we can't parse it for some reason, we just replace the string with a place holder.

Also there are two more exceptions that can be thrown in this chain that aren't handled.

A [SecurityException](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getProtectionDomain--) from getProtectionDomain(), a NullPointerExcetion from getLocation() if [getCodeSource()](https://docs.oracle.com/javase/8/docs/api/java/security/ProtectionDomain.html#getCodeSource--) returns null


In my opinion it might be better to either
1. Not automatically reflect the JAR location since it can throw an exception.
2. Swallow all of the exceptions and just replace with `<sample>.jar`

Reply:
agree. Replaced the JAR path reflection with `target/send-event-{version}-with-deps.jar`